notebooks/concepts/Fitting_multiprocess.ipynb

```diff
@@ -0,0 +1,1742 @@
+    "        x = parameters[0]\n",
+    "        y = parameters[1]\n",
+    "\n",
+    "        flux = self.cube[:, x, y] * funit\n",
+    "        sp = Spectrum1D(spectral_axis=self.wave, flux=flux)\n",
+    "        fitted_model = fit_lines(sp, self.model)\n",
+    "    \n",
+    "#         print(x, y, fitted_model.amplitude_0.value)\n",
+    "        \n",
+    "        return (x, y, fitted_model)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Multiprocess with worker pool"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Set up worker"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# (code repeated here for clarity).\n",
+    "region = SpectralRegion(6500*wunit, 7000*wunit)\n",
+    "spectrum = Spectrum1D(spectral_axis=wave, flux=flux3d[:, CENTER, CENTER] * funit)\n",
+    "i1, i2 = _to_edge_pixel(region.subregions[0], spectrum)\n",
+    "\n",
+    "wave_r = wave[i1:i2]\n",
+    "flux_cube_r = flux3d[i1:i2,:,:]\n",
+    "\n",
+    "# Multiprocessing worker.\n",
+    "worker = SpaxelWorker(flux_cube_r, wave_r, fitted_model_central)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Set up callback to collect results"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "2-D fit results (model parameters) come with different units associated to each 2-D array. So instead of a cube with fit results, we build a list of 2-D arrays storing `Quantity` instances.\n",
+    "\n",
+    "We work with `numpy` arrays throughout and only at the end build the list with results. This avoids the inneficiency of dealing with `Quantity` instances, constructors, and arrays, inside loops. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Build cube with empty slabs, one per model parameter. These store only parameter values \n",
+    "# for now, so a cube suffices.\n",
+    "result_cube = np.zeros(shape=(len(fitted_model_central.parameters), flux3d.shape[1], flux3d.shape[2]))\n",
+    "\n",
+    "# Callback to collect results from workers into the cube\n",
+    "def collect_result(result):\n",
+    "    x = result[0]\n",
+    "    y = result[1]\n",
+    "    model = result[2]\n",
+    "\n",
+    "    for index, name in enumerate(model.param_names):\n",
+    "        param = getattr(model, name)\n",
+    "        result_cube[index,x,y] = param.value"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Run pool"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Total time:  32.770500898361206 s\n",
+      "Time per spaxel: 0.03410041716790969 s\n"
+     ]
```

Comment:
It looks like there was only a factor of 3 speedup compared to serial processing.  How many cores were used?

Reply:
I tried with 8 and 7 cores.